### PR TITLE
Conductor commons

### PR DESCRIPTION
**Pull request by Jozefiel**
common logging
common workerImpl
common workflowImpl

### CODE REVIEW

File: frinx/common/conductor_enums.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    IN_PROGRESS = 'IN_PROGRESS'
+
+
+class WorkflowStatus(str, Enum):
+    COMPLETED = ('COMPLETED',)
+    FAILED = ('FAILED',)
+    PAUSED = ('PAUSED',)
+    RUNNING = ('RUNNING',)
+    TERMINATED = ('TERMINATED',)
+    TIMEOUT_OUT = ('TIMED_OUT',)
```
Comment (dav1d-b) [inline frinx/common/conductor_enums.py]: Why tuple if the enum type says string?

File: frinx/common/conductor_enums.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+class DoWhileEvaluatorType(str, Enum):
+    JAVASCRIPT = ('javascript',)
+    VALUE_PARAM = 'value-param'
+
+
+class RetryLogic(str, Enum):
+    FIXED = ('FIXED',)
+    EXPONENTIAL_BACKOFF = ('EXPONENTIAL_BACKOFF',)
+    LINEAR_BACKOFF = ('LINEAR_BACKOFF',)
```
Comment (dav1d-b) [inline frinx/common/conductor_enums.py]: Same as above

File: frinx/common/frinx_rest.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import os
+from types import MappingProxyType
+
+# RBAC CONFIGURATION
+x_tenant_id = os.getenv('X_TENANT_ID', 'frinx')
+x_from = os.getenv('X_FROM', 'fm-base-workers')
+x_auth_user_group = os.getenv('X_AUTH_USER_GROUP', 'network-admin')
+
+# SET SERVICES URL VARIABLES
+uniconfig_url_base = os.getenv('UNICONFIG_URL_BASE', 'http://uniconfig:8181/rests')
```
Comment (dav1d-b) [inline frinx/common/frinx_rest.py]: I would convert these variable names to uppercase, because they are constants

File: frinx/common/util.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    desc_representation: dict[str, Any] = {'description': description}
+    if labels:
+        desc_representation['labels'] = labels
+    if rbac:
+        desc_representation['rbac'] = rbac
+    output = json.dumps(desc_representation)
+    return output
+
+
+def parse_response(response: requests.Response) -> tuple[int, str]:
```
Comment (dav1d-b) [inline frinx/common/util.py]: This function is not used anywhere in the code.

File: frinx/common/worker/worker.py [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+class Config:
+    arbitrary_types_allowed = True
+    alias_generator = snake_to_camel_case
+    allow_population_by_field_name = True
+
+
+@dataclass(config=Config)
+class WorkerImpl(ABC):
+    task_def: TaskDefinition = None # type: ignore
+    task_def_template: type[BaseTaskdef] | type[DefaultTaskDefinition] = None # type: ignore
```
Comment (dav1d-b) [inline frinx/common/worker/worker.py]: None default value for a variable that should not allow None. We should fix the type issues in this class.

File: pyproject.toml [hunk 6]
```diff
@@ -60,6 +60,7 @@ exclude_lines = ['pragma: no cover',
 
 [tool.pyright]
 include = ['frinx']
+reportGeneralTypeIssues = "none"
```
Comment (dav1d-b) [inline pyproject.toml:63]: What is this configuration doing?

Comment (Jozefiel) [reply pyproject.toml:63]: Generate or suppress diagnostics for general type inconsistencies, unsupported operations, argument/parameter mismatches, etc. This covers all of the basic type-checking rules not covered by other rules. It does not include syntax errors. The default value for this setting is "error" but for that moment we suppress it.

File: frinx/common/util.py (hunk 4 above)
Comment (Jozefiel) [reply frinx/common/util.py]: removed from codebase

File: frinx/common/frinx_rest.py (hunk 3 above)
Comment (Jozefiel) [reply frinx/common/frinx_rest.py]: fixed

File: frinx/common/conductor_enums.py (hunk 1 above)
Comment (Jozefiel) [reply frinx/common/conductor_enums.py]: fixed

File: frinx/common/conductor_enums.py (hunk 2 above)
Comment (Jozefiel) [reply frinx/common/conductor_enums.py]: fixed

File: frinx/main.py
Comment (dav1d-b) [inline frinx/main.py:1]: Do we want this file?

Comment (Jozefiel) [reply frinx/main.py:1]: probably will be required later. for that moment can be deleted

**Review by dav1d-b — APPROVED** [review]

**Pull request merged**